torch/csrc/jit/python/init.cpp

```diff
@@ -1638,7 +1640,63 @@ void initJITBindings(PyObject* module) {
     }
     return type.value();
   });
-
+  py::enum_<SchemaArgType>(m, "_SchemaArgType")
+      .value("input", SchemaArgType::input)
+      .value("output", SchemaArgType::output);
+  py::class_<SchemaArgument>(m, "_SchemaArgument")
+      .def(py::init<SchemaArgType, size_t>())
+      .def_readwrite("type", &SchemaArgument::type)
+      .def_readwrite("index", &SchemaArgument::index);
+  py::class_<SchemaInfo>(m, "_SchemaInfo")
+      .def(py::init<FunctionSchema>())
+      .def("is_mutable", [](SchemaInfo& self) { return self.is_mutable(); })
+      .def(
+          "is_mutable",
+          [](SchemaInfo& self, size_t index) { return self.is_mutable(index); })
+      .def(
+          "is_mutable",
+          [](SchemaInfo& self, c10::string_view name) {
+            return self.is_mutable(name);
+          })
+      .def(
+          "may_alias",
+          [](SchemaInfo& self,
+             const SchemaArgument& lhs,
+             const SchemaArgument& rhs) { return self.may_alias(lhs, rhs); })
+      .def(
+          "may_contain_alias",
+          [](SchemaInfo& self,
+             const SchemaArgument& lhs,
+             const SchemaArgument& rhs) {
+            return self.may_contain_alias(lhs, rhs);
+          })
+      .def(
+          "add_argument_value",
+          [](SchemaInfo& self,
+             const std::string& name,
+             const py::object& value) {
+            if (name == "input") {
+              self.addArgumentValue("self", toTypeInferredIValue(value));
```

Comment:
Would it be okay to leave this logic on the python side since you already have this logic in schema check mode? Or is there a reason we need it here as well?

Reply:
It's necessary here due to how 'normalize_function' ends up naming the first argument when constructing a dictionary of argument names to argument values. Regardless of what the function's first argument name is supposed to be, 'normalize_function' will make the argument name "input." 

To match this, the standardize_name() function on the python side changes all names of "self" into "input." 

However, in this add_argument_value, we need the original name of the argument, not the normalized name which is always "input," thus this check exists here so that extra processing doesn't need to occur on the python side.